File: backend/src/main/java/com/ifpb/agendapet/user/dto/UserResponseDTO.java
package com.ifpb.agendapet.user.dto;

import com.ifpb.agendapet.shared.enums.RoleEnum;
import com.ifpb.agendapet.shared.enums.GenderEnum;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Set;
import java.util.UUID;

public record UserResponseDTO(UUID id, String name, String email, String avatarUrl, GenderEnum gender, LocalDate birthday, Set<RoleEnum> roles, LocalDateTime created_at) {
}

File: frontend/src/app/models/DTO/user-response-DTO.ts
export interface UserResponseDTO {
  id: string;
  name: string;
  email: string;
  avatarUrl?: string;
  gender: GenderEnum;
  birthday: string;
  roles: RoleEnum[];
  created_at: string;
}

export enum GenderEnum {
  MALE = 'MALE',
  FEMALE = 'FEMALE',
  OTHER = 'OTHER',
}

export enum RoleEnum {
  ADMIN = 'ADMIN',
  DOCTOR = 'DOCTOR',
  TUTOR = 'TUTOR',
  RECEPTIONIST = 'RECEPTIONIST',
}
